### DISCUSSION TITLE
Zustand store update and race conditions

### DISCUSSION THREAD

**Discussion by hexwit** [pmndrs/zustand#2034] [General]
Hello!

Business logic of my app requires 2 http requests at application start. 
Both of them updates different parts of the store. Those requests issued at the same time, in parallel. And I don't know which one finish first. 

So my question is this: is it safe to call methods that update zustand's store from async functions? 
Is there any chance to get race conditions? 

P.S. I am not aware about zustand internals, assume that it is possible that zustand has some async logic.

**Comment by dai-shi** [comment]
> P.S. I am not aware about zustand internals, assume that it is possible that zustand has some async logic.

Reading the internals might help you understand the behavior: https://twitter.com/dai_shi/status/1583082766081531905
